### ISSUE TITLE
Maya: OP Menu Redesign Proposal - Better usability/ User experience 

### ISSUE THREAD

**Issue by LiborBatek** [ynput/OpenPype#4285]
As been working with OpenPype for some time it bothers me inconsistencies in GUI and terminology...should be addressed imho.

Here is an example of different terminology accross OP entities:
![UI_Maya_Proposal_01a](https://user-images.githubusercontent.com/112623825/211011885-1bd9faca-a51e-45c5-9cb1-1f302108a703.jpg)

I also propose make a redesign in the menu itself for better usability.

1) Most important/frequent actions at the top of the menu
2) Logical grouping of those actions according to user interaction
3) Hiearchical listing of menu items into sub groups (sub menus) lessening overall clutter to minimum - better readibility.

First draft here:

![UI_Maya_Proposal_01b](https://user-images.githubusercontent.com/112623825/211012353-da600d65-7f05-4f74-95c5-294dbfc7b648.jpg)

**Comment by tokejepsen** [comment]
Nice yeah!

If we are redesigning, maybe get rid of the `...` at the end of labels?

What is `Assets...`?

Think `Looks...` should be `Look Manager`.

Not sure about `Create Asset`/`Publish Asset`/`Manage Assets`. Is it consistent with other DCCs?

Also not sure `Library` should be demoted to a submenu (?).

**Comment by BigRoy** [comment]
> If we are redesigning, maybe get rid of the ... at the end of labels?

---

Menus often show that when clicking the menu entry would show a pop-up or dialog instead of having an instant effect. But I don't have a preference over whether to keep or remove them.

---

Utilities tends to be a name or category that basically doesn't say much about what's in the menu. I'd say that if Library Loader is moved out of there it might be worth avoiding the Experimental Tools menu entry and just pushing the Generate first workfile + template builder into clearer named submenu of its own since I believe they are both related to building those workfile templates?

It actually seems that has already happened since:
![afbeelding](https://user-images.githubusercontent.com/2439881/211041476-1f74cab2-4ce9-4302-8488-35664cecfef6.png)

---

I'd pick Submenus only when you now the entries in the menu will start to increase a bit more. Currently the size isn't that huge that I feel it reall warrants having extra menus. Instead I'd argue to maybe but dividers with labels in the menus, kind of like in this screenshot:
![afbeelding](https://user-images.githubusercontent.com/2439881/211041370-d66f14df-ce2b-4a1e-9920-a23fcf6df285.png)

---

tl;dr in short I think going from this screenshot:
![afbeelding](https://user-images.githubusercontent.com/2439881/211041610-c82eeff8-9776-4fe5-860b-1e22b1dc6ae6.png)

To move Work Files like you've done and maybe just add a descriptive header to the dividers like "Set defaults" but still keep the entries for now directly in that menu - I think.

**Comment by LiborBatek** [comment]
@BigRoy  dividers with labels sound great! 

to the rest, yeah why not, I dont prefer those ```Name...``` indicating something will roll out, lets loose them!

Also those ```Utilities``` I wasnt so sure how to name em and also ```Library``` obviously relates more to ```Asset``` so should be close to them.

I have biggest issue with non logical placement of those menu items...they just sit here and there and arent logically placed...

Most of the time as a user you first go to open workfile, then loading some asset and apply some look. 
Second most common tasks are ```create``` and ```publish```.

The rest is not so commonly used and can be lower in the OP menu.

Thanks for your thoughts guys!

**Comment by LiborBatek** [comment]
I think as major release of OP 3.16 is comming close with conjuction of AYON and new Publisher for Maya host. Its good time to make those changes happen!

Here is another pass of redesign of OP menu elements which are distilled from above discussion.

First I think its good idea to loose ```Experimental tools``` item or at least to move it away from main OP menu to ```OpenPype Tools``` menu.
![OP_Menu_adj01](https://github.com/ynput/OpenPype/assets/112623825/81d0fbcb-7926-4981-a039-82dbe0e1f9ec)

Second some more changes I made mostly originating on @mkolar ideas...

**Comment by LiborBatek** [comment]
@antirotor @m-u-r-p-h-y  what u think about proposal for new menu layout? Would be glad to hear your opinions too...

**Comment by LiborBatek** [comment]
@m-u-r-p-h-y  I am strongly against alphabetical sorting....

there is actualy workflow routine as

1. Workfiles (first always)
2. Loading Assets (usually)
3. Publish Assets (a lot of times as third or even second action could be #2)

Then there are tools for managing the workfile scene as

Manage / Inventory
Looks / Looks Assigner

then there are optional tools like

Set Defaults
Template builder
etc

Regarding naming...I was using Milans ones mostly

**Comment by LiborBatek** [comment]
Ok I have updated the OP Menu after @m-u-r-p-h-y points... 

the simpler the better
![OP_Menu_Compare_v03](https://github.com/ynput/OpenPype/assets/112623825/ba48a7ff-17a5-4b58-90ee-97ca8a219887)

I have an idea / variant speaking of ```Manage``` separate category in my head...will post it soon

**Comment by m-u-r-p-h-y** [comment]
I do not agree with you at all. Looking for an item in menu is not based on workflow. Items should be logically grouped together and then sorted alphabetically to help find things quickly. 

If you consider a modeling task with 30 workfiles, you only start with the Workflie tool for the first one and not for the other 29 cases. You are ending with Workfile tool most often than starting.

Look at Maya menu 
![image](https://github.com/ynput/OpenPype/assets/352795/0bacf6fa-8b87-4b1c-8f26-8a19ab0adc08)

**Comment by LiborBatek** [comment]
> If you consider a modeling task with 30 workfiles, you only start with the Workflie tool for the first one and not for the other 29 cases. You are ending with Workfile tool most often than starting.

It is like saying you do not start by ```File>Open```. 
If I can name one single item I hit most in OP Menu than it is ```Workfiles``` ...its just a fact. 

How about when switching context?? You dont use that for it never?????? I dont think so :)

**Comment by LiborBatek** [comment]
Here is another mockup version 04.

Not sure if better then previous but take it as a food for thought...

![OP_Menu_New_v04](https://github.com/ynput/OpenPype/assets/112623825/d21c43e4-1357-4c2d-8e9b-de45be7cc2bd)

I personally think the previous version (call it v03) is probably better imho.

**Comment by m-u-r-p-h-y** [comment]
again, the order of menu items is not driven by the number of clicks. It is about the convenience of finding it. Items in each group should be sorted alphabetically regardless of their virtual importance.
Separating Workfile tool to a separate group is a valid point (even though I do not think it is necessary)

**Comment by m-u-r-p-h-y** [comment]
For example:
![MayaMenu3](https://github.com/ynput/OpenPype/assets/352795/e24dff20-4af2-4ec3-8740-9eb7fcb31d97)

**Comment by BigRoy** [comment]
I personally prefer the old "Load..", "Manage.." similar to "Open Scene...", or "Open.." as you regularly see in menus. To me "Loader..." doesn't tell me more. It also feels weird to have the `...` after something that's the name of the tool.

Otherwise I'd also expect "Placeholder creator".

By the way, the last few designs are missing the "Create..." entry. (which in practice is also the Publisher, but there's been confusion in practice with "Create" removed in the menus plus it's a nice quick way to only access the create page.) That's another 👍 to go for Create... and Publish... instead of Publisher...

**Comment by LiborBatek** [comment]
> Here is another mockup version 04.
> 
> Not sure if better then previous but take it as a food for thought...
> 
> ![OP_Menu_New_v04](https://user-images.githubusercontent.com/112623825/250071962-d21c43e4-1357-4c2d-8e9b-de45be7cc2bd.png)
> 

@mkolar any thoughts on this?
